plugins/aea-ledger-ethereum/tests/test_ethereum.py

```diff
@@ -920,3 +980,91 @@ def test_gas_estimation(
+    ethereum_api = EthereumApi(**get_default_gas_strategies())
+    ethereum_api._is_gas_estimation_enabled = True
+    args = {
+        "sender_address": ec1.address,
+        "destination_address": ec2.address,
+        "amount": 1,
+        "tx_fee": 0,
+        "tx_nonce": "",
+        "max_fee_per_gas": None,
+    }
+    with patch.object(
+        ethereum_api, "try_get_gas_pricing", return_value={"gasPrice": 12}
+    ) as get_gas_pricing:
+        result = ethereum_api.get_transfer_transaction(**args)
+        assert result["gasPrice"] == 12
+        get_gas_pricing.assert_called_once()
+
+    args["gas_price"] = 13
+    result = ethereum_api.get_transfer_transaction(**args)
+    assert result["gasPrice"] == 13
+
+
+def test_get_base_fee_multiplier() -> None:
+    """Test get_base_fee_multiplier."""
+    assert get_base_fee_multiplier(35) == 2.0
+    assert get_base_fee_multiplier(45) == 1.6
+    assert get_base_fee_multiplier(105) == 1.4
+    assert get_base_fee_multiplier(205) == 1.2
+
+
+def test_estimate_priority_fee() -> None:
+    """Test estimate_priority_fee."""
+    # return none on no rewards
+    web3_mock = Mock()
+    web3_mock.eth.fee_history = Mock(return_value={"reward": []})
+    assert estimate_priority_fee(web3_mock, 1, 1, 1, 1, 11, 1, 1) is None
+
+    # If we have big increase in value, we could be considering "outliers" in our estimate
+    # Skip the low elements and take a new median
+    web3_mock.eth.fee_history = Mock(return_value={"reward": [[1], [10], [10000]]})
+    assert estimate_priority_fee(web3_mock, 1, 1, 1, 1, 11, 1, 1) == 10000
+
+
+def test_try_get_revert_reason_http_error_propagated(ethereum_testnet_config) -> None:
+    """Test httperror reraised if get_revert_reason fails."""
+    eth_api = EthereumApi(**ethereum_testnet_config)
+    tx = {"from": 1, "to": 1, "input": 1, "value": 1, "blockNumber": 1}
+    api_mock = Mock()
+    api_mock.eth.call = Mock(side_effect=HTTPError("http_error"))
+    with patch.object(eth_api, "_api", new=api_mock):
+        with pytest.raises(HTTPError):
+            eth_api._try_get_revert_reason(tx, raise_on_try=True)
+
+
+def test_get_gas_price_strategy() -> None:
+    """Test get_gas_price_strategy and strategies."""
+    strategy = get_gas_price_strategy(None)
+    assert strategy is rpc_gas_price_strategy_wrapper
+    assert "gasPrice" in strategy(Mock(), Mock())
+
+    strategy = get_gas_price_strategy_eip1559_polygon("test", {"gasPrice": 12})
+
+    resp_mock = Mock()
+    resp_mock.status_code = 300
+    with patch("aea.helpers.http_requests.get", return_value=resp_mock):
+        assert strategy(Mock(), Mock()) == {"gasPrice": 12}
+
+    resp_mock.status_code = 200
+    resp_mock.json = Mock(return_value={"fast": {"maxFee": 2, "maxPriorityFee": 2}})
+    with patch("aea.helpers.http_requests.get", return_value=resp_mock):
+        assert strategy(Mock(), Mock()) == {
+            "maxFeePerGas": 2000000000,
+            "maxPriorityFeePerGas": 2000000000,
```

Comment:
minor suggestion for readability; `int(2e9)` (personally find this easiest to read) or `2_000_000_000`